libsql-server/tests/auth/mod.rs

```diff
@@ -40,16 +40,34 @@ async fn make_standalone_server() -> Result<(), Box<dyn std::error::Error>> {
     Ok(())
 }
 
+fn gen_test_jwt_auth() -> (Auth, String) {
+    let doc = Ed25519KeyPair::generate_pkcs8(&ring::rand::SystemRandom::new()).unwrap();
+    let encoding_key = EncodingKey::from_ed_der(doc.as_ref());
+
+    let pair = Ed25519KeyPair::from_pkcs8(doc.as_ref()).unwrap();
+    let decoding_key = DecodingKey::from_ed_der(pair.public_key().as_ref());
+
+    let claims = Token::default();
+
+    let header = jsonwebtoken::Header::new(jsonwebtoken::Algorithm::EdDSA);
+    let token = jsonwebtoken::encode(&header, &claims, &encoding_key).unwrap();
+
+    let jwt_keys = vec![decoding_key];
+
+    let auth = Auth::new(user_auth_strategies::Jwt::new(jwt_keys));
+
+    (auth, token)
+}
+
```

Comment:
there's already code for that in `common:auth`

Reply:
common auth doesn't provide a decoding key required to create the auth.

Reply:
I don't think it makes sense to add to that code and this code is quite trivial It hink its okay to keep this here.

Reply:
I wrote this code ealier today debugging for Jamie:
```
diff --git a/libsql-server/src/auth/authorized.rs b/libsql-server/src/auth/authorized.rs
index dc3a06cae2..6eb8913f56 100644
--- a/libsql-server/src/auth/authorized.rs
+++ b/libsql-server/src/auth/authorized.rs
@@ -239,4 +239,27 @@ mod test {
         assert!(!auth.has_right(Scope::Namespace("ns".into()), Permission::Read));
         assert!(!auth.has_right(Scope::Namespace("ns".into()), Permission::Write));
     }
+
+    #[test]
+    fn bug() {
+        std::env::set_var("LIBSQL_GID", "my_group");
+        let auth = Authorized {
+            read_only_attach: Some(Scopes {
+                namespaces: Some([NamespaceName::from("foo")].into_iter().collect()),
+                ..Default::default()
+            }),
+            read_write: Some(Scopes {
+                tags: Some([String::from("my_group")].into_iter().collect()),
+                ..Default::default()
+            }),
+            ..Default::default()
+        };
+
+        assert!(auth.has_right(Scope::Namespace("foo".into()), Permission::AttachRead));
+        assert!(auth.has_right(Scope::Namespace("foo".into()), Permission::Read));
+        assert!(auth.has_right(Scope::Namespace("foo".into()), Permission::Write));
+        assert!(!auth.has_right(Scope::Namespace("bar".into()), Permission::AttachRead));
+        assert!(auth.has_right(Scope::Namespace("bar".into()), Permission::Read));
+        assert!(auth.has_right(Scope::Namespace("bar".into()), Permission::Write));
+    }
 }
diff --git a/libsql-server/tests/auth/jwt_key.pem b/libsql-server/tests/auth/jwt_key.pem
deleted file mode 100644
index ac9591cf12..0000000000
--- a/libsql-server/tests/auth/jwt_key.pem
+++ /dev/null
@@ -1,3 +0,0 @@
------BEGIN PUBLIC KEY-----
-MCowBQYDK2VwAyEAbz/oEg1rMRGY12X4Q0GXioX1hXaM69o9kp7h0eCLD/E=
------END PUBLIC KEY-----
diff --git a/libsql-server/tests/auth/mod.rs b/libsql-server/tests/auth/mod.rs
index 5834d44c33..6d6f407d54 100644
--- a/libsql-server/tests/auth/mod.rs
+++ b/libsql-server/tests/auth/mod.rs
@@ -3,10 +3,8 @@
 
 use futures::SinkExt as _;
 use libsql::Database;
-use libsql_server::{
-    auth::{user_auth_strategies, Auth},
-    config::UserApiConfig,
-};
+use libsql_server::auth::{user_auth_strategies, Auth};
+use libsql_server::config::{AdminApiConfig, UserApiConfig};
 use tempfile::tempdir;
 use tokio_stream::StreamExt;
 use tokio_tungstenite::{
@@ -15,13 +13,12 @@ use tokio_tungstenite::{
 };
 use turmoil::net::TcpStream;
 
-use crate::common::net::{init_tracing, SimServer, TestServer, TurmoilConnector};
-
-const TEST_JWT_KEY: &str = "eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MjE2NTIwNTB9.5XhUDHQhtShszTssjjUzVuJA3r-031mT4inVkvYEYz64sOCxnNpZUZdVF-CmZ4t-JTSXFlm8ddscBgkhccBxDg";
+use crate::common::auth::{encode, key_pair};
+use crate::common::http::Client;
+use crate::common::net::{init_tracing, SimServer, TestServer, TurmoilAcceptor, TurmoilConnector};
 
-async fn make_standalone_server() -> Result<(), Box<dyn std::error::Error>> {
-    let jwt_pem = include_bytes!("jwt_key.pem");
-    let jwt_keys = vec![jsonwebtoken::DecodingKey::from_ed_pem(jwt_pem).unwrap()];
+async fn make_standalone_server(key: String) -> Result<(), Box<dyn std::error::Error>> {
+    let jwt_keys = vec![jsonwebtoken::DecodingKey::from_ed_components(&key).unwrap()];
 
     init_tracing();
     let tmp = tempdir()?;
@@ -32,6 +29,12 @@ async fn make_standalone_server() -> Result<(), Box<dyn std::error::Error>> {
             auth_strategy: Auth::new(user_auth_strategies::Jwt::new(jwt_keys)),
             ..Default::default()
         },
+        admin_api_config: Some(AdminApiConfig {
+            acceptor: TurmoilAcceptor::bind(([0, 0, 0, 0], 9090)).await.unwrap(),
+            connector: TurmoilConnector,
+            disable_metrics: true,
+        }),
+        disable_namespaces: false,
         ..Default::default()
     };
 
@@ -42,14 +45,16 @@ async fn make_standalone_server() -> Result<(), Box<dyn std::error::Error>> {
 
 #[test]
 fn http_hrana() {
+    let (encoding, decoding) = key_pair();
     let mut sim = turmoil::Builder::new().build();
-    sim.host("primary", make_standalone_server);
+    let key = encode(&serde_json::json!({}), &encoding);
+    sim.host("primary", move || {
+        let decoding = decoding.clone();
+        async move { make_standalone_server(decoding).await }
+    });
     sim.client("client", async {
-        let db = Database::open_remote_with_connector(
-            "http://primary:8080",
-            TEST_JWT_KEY,
-            TurmoilConnector,
-        )?;
+        let db =
+            Database::open_remote_with_connector("http://primary:8080", key, TurmoilConnector)?;
         let conn = db.connect()?;
 
         conn.execute("create table t(x text)", ()).await?;
@@ -65,15 +70,21 @@ fn embedded_replica() {
     let tmp_embedded = tempdir().unwrap();
     let tmp_embedded_path = tmp_embedded.path().to_owned();
 
+    let (encoding, decoding) = key_pair();
+    let key = encode(&serde_json::json!({}), &encoding);
+
     let mut sim = turmoil::Builder::new().build();
-    sim.host("primary", make_standalone_server);
+    sim.host("primary", move || {
+        let decoding = decoding.clone();
+        async move { make_standalone_server(decoding).await }
+    });
     sim.client("client", async move {
         let path = tmp_embedded_path.join("embedded");
 
         let db = Database::open_with_remote_sync_connector(
             path.to_str().unwrap(),
             "http://primary:8080",
-            TEST_JWT_KEY,
+            key,
             TurmoilConnector,
             false,
             None,
@@ -93,8 +104,14 @@ fn embedded_replica() {
 #[test]
 fn ws_hrana() {
     let mut sim = turmoil::Builder::new().build();
-    sim.host("primary", make_standalone_server);
-    sim.client("client", async {
+    let (encoding, decoding) = key_pair();
+    let key = encode(&serde_json::json!({}), &encoding);
+
+    sim.host("primary", move || {
+        let decoding = decoding.clone();
+        async move { make_standalone_server(decoding).await }
+    });
+    sim.client("client", async move {
         let url = "ws://primary:8080";
 
         let req = url.into_client_request().unwrap();
@@ -116,7 +133,7 @@ fn ws_hrana() {
         }
 
         let msg = ClientMsg::Hello {
-            jwt: Some(TEST_JWT_KEY.to_string()),
+            jwt: Some(key.clone()),
         };
 
         let msg_data = serde_json::to_string(&msg).unwrap();
@@ -134,3 +151,75 @@ fn ws_hrana() {
 
     sim.run().unwrap();
 }
+
+#[test]
+fn test_auth_bug() {
+    std::env::set_var("LIBSQL_GID", "my_group");
+
+    let (encoding, decoding) = key_pair();
+
+    let mut sim = turmoil::Builder::new().build();
+    sim.host("primary", move || {
+        let decoding = decoding.clone();
+        async move { make_standalone_server(decoding).await }
+    });
+
+    sim.client("client", async move {
+        let claims = serde_json::json!({
+            "iat": 1721566104,
+            "p": {
+                "roa": {
+                    "ns": [
+                        "foo"
+                    ]
+                },
+                "rw": {
+                    "tags": [
+                        "my_group"
+                    ]
+                }
+            }
+        });
+
+        let key = encode(&claims, &encoding);
+        let client = Client::new();
+        client
+            .post(
+                "http://primary:9090/v1/namespaces/foo/create",
+                &serde_json::json!({}),
+            )
+            .await
+            .unwrap();
+        client
+            .post(
+                "http://primary:9090/v1/namespaces/bar/create",
+                &serde_json::json!({}),
+            )
+            .await
+            .unwrap();
+
+        let foo_db = Database::open_remote_with_connector(
+            "http://foo.primary:8080",
+            &key,
+            TurmoilConnector,
+        )?;
+
+        let foo_conn = foo_db.connect()?;
+
+        foo_conn.execute("create table t(x text)", ()).await?;
+
+        let bar_db = Database::open_remote_with_connector(
+            "http://bar.primary:8080",
+            &key,
+            TurmoilConnector,
+        )?;
+
+        let bar_conn = bar_db.connect()?;
+
+        bar_conn.execute("create table t(x text)", ()).await?;
+
+        Ok(())
+    });
+
+    sim.run().unwrap();
+}
```

Reply:
your function is almost verbatim the function in `common::auth`

Reply:
https://github.com/tursodatabase/libsql/blob/main/libsql-server/tests/common/auth.rs

Reply:
Oh I misread what that string return was returning...okay I can fix this in a follow up we shouldn't block on this

Reply:
https://github.com/tursodatabase/libsql/pull/1586